src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -1666,7 +1666,8 @@ void EmitNode(RegexNode node, RegexNode? subsequent = null, bool emitLengthCheck
 
                 // Put the node's code into its own scope. If the node contains labels that may need to
                 // be visible outside of its scope, the scope is still emitted for clarity but is commented out.
-                using (EmitScope(writer, DescribeNode(node), faux: PossiblyBacktracks(node) && !node.IsAtomicByParent()))
+                string capName = ExtractCapName(node, rm.Code.Tree.CapNames);
```

Comment:
We should only do this inside of DescribeNode in the case where the node is actually a capture.  I suggest adding a DescribeCapture helper function that DescribeNode uses; it is then responsible for formatting the capture description, including looking up any name that's needed for the rendering.